components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -259,31 +285,80 @@ int GetNormal(int min, int max) {
   return min + floor((max - min) * GetNormal());
 }
 
-using GetWeighting = double(const ArticleWeight& weight);
+using GetWeighting = std::function<double(const mojom::FeedItemMetadataPtr& metadata, const ArticleWeight& weight)>;
```

Comment:
What can we use instead of `std::function` (banned) that allows extra arguments to be captured in a lambda? we had some issues with having a simple function here (just `double(const mojom::FeedItemMetadataPtr& metadata, const ArticleWeight& weight)`)